packages/flutter_tools/test/build_system/targets/dart_test.dart

```diff
@@ -9,85 +9,164 @@ import 'package:flutter_tools/src/build_info.dart';
 
 void main() {
   group('dart rules', () {
     Testbed testbed;
     BuildSystem buildSystem;
-    Environment environment;
+    Environment androidEnvironment;
+    Environment iosEnvironment;
+    MockProcessManager mockProcessManager;
+
+    setUpAll(() {
+      Cache.disableLocking();
+    });
 
     setUp(() {
+      mockProcessManager = MockProcessManager();
       testbed = Testbed(setup: () {
-        final Directory cacheDir = fs.currentDirectory.childDirectory('cache');
-        environment = Environment(
+        androidEnvironment = Environment(
+          projectDir: fs.currentDirectory,
+          defines: <String, String>{
+            kBuildMode: getNameForBuildMode(BuildMode.profile),
+            kTargetPlatform: getNameForTargetPlatform(TargetPlatform.android_arm),
+          }
+        );
+        iosEnvironment = Environment(
           projectDir: fs.currentDirectory,
-          cacheDir: cacheDir,
           defines: <String, String>{
             kBuildMode: getNameForBuildMode(BuildMode.profile),
+            kTargetPlatform: getNameForTargetPlatform(TargetPlatform.ios),
           }
         );
-        buildSystem = BuildSystem(<String, Target>{
-          kernelSnapshot.name: kernelSnapshot,
-          aotElf.name: aotElf,
-        });
-        final String skyEngineLine = platform.isWindows
-            ? r'sky_engine:file:///C:/cache/pkg/sky_engine/lib/'
-            : 'sky_engine:file:///cache/pkg/sky_engine/lib/';
+        buildSystem = BuildSystem();
+        HostPlatform hostPlatform;
+        if (platform.isWindows) {
+          hostPlatform = HostPlatform.windows_x64;
+        } else if (platform.isLinux) {
+          hostPlatform = HostPlatform.linux_x64;
+        } else if (platform.isMacOS) {
+           hostPlatform = HostPlatform.darwin_x64;
+        } else {
+          assert(false);
+        }
+         final String skyEngineLine = platform.isWindows
+            ? r'sky_engine:file:///C:/bin/cache/pkg/sky_engine/lib/'
+            : 'sky_engine:file:///bin/cache/pkg/sky_engine/lib/';
         fs.file('.packages')
           ..createSync()
           ..writeAsStringSync('''
 # Generated
 $skyEngineLine
 flutter_tools:lib/''');
-        fs.file(fs.path.join(cacheDir.path, 'pkg', 'sky_engine', 'lib', 'ui', 'ui.dart')).createSync(recursive: true);
-        fs.file(fs.path.join(cacheDir.path, 'pkg', 'sky_engine', 'sdk_ext', 'vmservice_io.dart')).createSync(recursive: true);
+        fs.file(fs.path.join('bin', 'cache', 'pkg', 'sky_engine', 'lib', 'ui', 'ui.dart')).createSync(recursive: true);
+        fs.file(fs.path.join('bin', 'cache', 'pkg', 'sky_engine', 'sdk_ext', 'vmservice_io.dart')).createSync(recursive: true);
+        fs.file(fs.path.join('bin', 'cache', 'dart-sdk', 'bin', 'dart')).createSync(recursive: true);
+        fs.file(fs.path.join('bin', 'cache', 'artifacts', 'engine', getNameForHostPlatform(hostPlatform), 'frontend_server.dart.snapshot')).createSync(recursive: true);
```

Comment:
These lines are getting a bit long.

Reply:
Tried to shorten most of them. Will look for a better solution for the testbed though.